GW2EIEvtcParser/EIData/CombatReplay/CombatReplay.cs

```diff
@@ -241,24 +265,64 @@ internal void AddOverheadIcons(IEnumerable<Segment> segments, AbstractSingleActo
         {
-            IReadOnlyList<EffectEvent> allEffectEvents = log.CombatData.GetEffectEvents().Where(x => !knownEffectIDs.Contains(x.EffectID) && !x.Src.GetFinalMaster().IsPlayer && (!x.IsAroundDst || !x.Dst.GetFinalMaster().IsPlayer) && x.Time >= start && x.Time <= end && x.EffectID > 0).ToList(); ;
-            var effectGUIDs = allEffectEvents.Select(x => log.CombatData.GetEffectGUIDEvent(x.EffectID).ContentGUID).ToList();
-            var effectGUIDsDistinct = effectGUIDs.GroupBy(x => x).ToDictionary(x => x.Key, x => x.ToList().Count);
-            foreach (EffectEvent effectEvt in allEffectEvents)
+            int tetherStart = 0;
+            AbstractSingleActor actor = null;
+            foreach (AbstractBuffEvent fixation in fixations)
             {
-                if (effectEvt.IsAroundDst)
+                if (fixation is BuffApplyEvent)
                 {
-                    replay.Decorations.Insert(0, new CircleDecoration(true, 0, 180, ((int)effectEvt.Time, (int)effectEvt.Time + 100), "rgba(0, 255, 255, 0.5)", new AgentConnector(log.FindActor(effectEvt.Dst))));
+                    tetherStart = (int)fixation.Time;
+                    actor = npcs.FirstOrDefault(x => x.AgentItem == fixation.CreditedBy);
                 }
                 else
                 {
-
-                    replay.Decorations.Insert(0, new CircleDecoration(true, 0, 180, ((int)effectEvt.Time, (int)effectEvt.Time + 100), "rgba(0, 255, 255, 0.5)", new PositionConnector(effectEvt.Position)));
+                    int tetherEnd = (int)fixation.Time;
+                    if (actor != null)
+                    {
+                        Decorations.Add(new LineDecoration(0, (tetherStart, tetherEnd), color, new AgentConnector(player), new AgentConnector(actor)));
+                    }
                 }
             }
+        }
 
+        /// <summary>
+        /// Add tether decorations from the <paramref name="player"/> to <paramref name="npcs"/>. <br></br>
+        /// Uses <see cref="List{T}"/> of <see cref="AbstractBuffEvent"/> to find the buff owner and attach the two agents.
+        /// </summary>
+        /// <param name="fixations">Buff events of the fixations.</param>
+        /// <param name="npcs">NPCs targetting the player.</param>
+        /// <param name="player">Player target of the tether.</param>
+        /// <param name="color">Color of the tether.</param>
+        internal void AddPlayerToNPCTethering(List<AbstractBuffEvent> fixations, IReadOnlyList<AbstractSingleActor> npcs, AbstractPlayer player, string color)
+        {
+            int tetherStart = 0;
+            AbstractSingleActor actor = null;
+            foreach (AbstractBuffEvent fixation in fixations)
+            {
+                if (fixation is BuffApplyEvent)
+                {
+                    tetherStart = (int)fixation.Time;
+                    actor = npcs.FirstOrDefault(x => x.AgentItem == fixation.CreditedBy);
+                }
+                else
```

Comment:
This should be "else if (fixation is BuffRemoveAllEvent)", it was fine when there was not a method as GetFilteredList can only return a List of BuffApply/BuffRemoveAll events but with a generic method additional checks should be done as the input is a list of AbstractBuffEvent.